djstripe/models.py

```diff
@@ -741,7 +725,7 @@ def sync_from_stripe_data(cls, stripe_invoice, send_receipt=True):
             invoice.period_end = period_end
 
             if item.get("plan"):
```

Comment:
Since plan is now a foreign key, you have to search for all the instances where plan is set to a string. For instance, this conditional should read:
```python
    if item.get("plan"):
        plan = Plan.objects.get(stripe_id=item["plan"]["id"])
    else:
        plan = None
```